### PR TITLE
HitBTC: Add Subscription Configuration

### PR DESCRIPTION
**Pull request by gbjk**
## HitBTC

* Add Subscription configuration
* Upgrade test configuration

## New Library Dependencies

- [sprig](github.com/Masterminds/sprig/v3)   ( Also included in 4 other template PRs )

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: exchanges/hitbtc/hitbtc_websocket.go [hunk 1]
```diff
@@ -31,6 +34,22 @@ const (
...
+	subscription.OrderbookChannel: "Orderbook",
+	subscription.CandlesChannel:   "Candles",
+	subscription.AllTradesChannel: "Trades",
+	subscription.MyAccountChannel: "Reports",
+}
+
+var defaultSubscriptions = subscription.List{
+	{Enabled: true, Asset: asset.Spot, Channel: subscription.TickerChannel},
+	{Enabled: true, Asset: asset.Spot, Channel: subscription.OrderbookChannel},
+	{Enabled: true, Asset: asset.Spot, Channel: subscription.CandlesChannel, Interval: kline.ThirtyMin, Levels: 100},
```
Comment (gloriousCode) [inline exchanges/hitbtc/hitbtc_websocket.go:48]:
```
[WARN]  | WEBSOCKET | 05/12/2024 16:07:15 | HitBTC - Unhandled websocket message: {"jsonrpc":"2.0","method":"snapshotCandles",
...
```
Not your ~keys~ bug not your ~coins~ fix

File: exchanges/hitbtc/hitbtc_websocket.go [hunk 2]
```diff
@@ -838,3 +807,50 @@ func (h *HitBTC) wsGetTrades(c currency.Pair, limit int64, sort, by string) (*Ws
 	}
 	return &response, nil
 }
+
+// subToReq returns the subscription as a map to populate WsRequest
+func subToReq(s *subscription.Subscription, maybePair ...currency.Pair) *WsRequest {
```
Comment (gloriousCode) [inline exchanges/hitbtc/hitbtc_websocket.go:812]: [Bithumb](https://github.com/thrasher-corp/gocryptotrader/pull/1618/files#diff-4d15fff6eb93a58ccc68b3ed41eec1a75dc0b3bb5bd2daba9dc7f1d2b2aa1e0aR96) got a test for this func, I request one here too if possible

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Nice, tested auth and unauth. just one thing

File: exchanges/hitbtc/hitbtc_websocket.go (hunk 1 above)
Comment (gbjk) [reply exchanges/hitbtc/hitbtc_websocket.go:48]: Spun off into a miniseries #1740

File: exchanges/hitbtc/hitbtc_websocket.go (hunk 2 above)
Comment (gbjk) [reply exchanges/hitbtc/hitbtc_websocket.go:812]: Fixed https://github.com/gbjk/gocryptotrader/commit/1bd1ef2dc

**Review by gloriousCode — APPROVED** [review]
tACK!

File: cmd/documentation/exchanges_templates/hitbtc.tmpl [hunk 3]
```diff
@@ -93,12 +93,24 @@ if err != nil {
...
 
-```go
-	// Exchanges will be abstracted out in further updates and examples will be
-	// supplied then
-```
+Subscriptions are for [v2 api](https://hitbtc-com.github.io/hitbtc-api/#socket-api-reference)
+
+All subscriptions are for spot.
+
+Default Public Subscriptions:
```
Comment (shazbert) [inline cmd/documentation/exchanges_templates/hitbtc.tmpl:102]: NOTE 4 ME: Might be able to make this entire change dynamic across all exchanges in #1619 reduce implementation toil.

**Review by shazbert — APPROVED** [review]
Thanks 🖖 tACK.

**Review by thrasher- — APPROVED** [review]
tACK public subs, utACK private. Working well!

**Pull request merged**